src/index.ts

```diff
@@ -11,42 +14,37 @@ function transpileToCommonJS(code: string) {
   })
 }
 
-/** jest transformers must export a process function
- * @param {src} code to transform
- * @param {path} the path to the file
- */
-export function process(src: string, path: string) {
-  const { descriptor } = parse(src)
-
-  let bindings: BindingMetadata
-  let scriptSetupResult: SFCScriptBlock
-
-  if (descriptor.script || descriptor.scriptSetup) {
-    scriptSetupResult = compileScript(descriptor)
-    bindings = scriptSetupResult.bindings
-  }
-
-  const template = compileTemplate({
-    filename: descriptor.filename,
-    source: descriptor.template.content,
-    compilerOptions: {
-      bindingMetadata: bindings
+class VueTsJestTransformer extends TsJestTransformer implements Transformer {
+  process(input: string, _filePath: Config.Path, _jestConfig: Config.ProjectConfig, _transformOptions?: TransformOptions): TransformedSource | string {
+    const { descriptor } = parse(input)
+    let bindings: BindingMetadata | undefined
+    let scriptSetupResult: SFCScriptBlock
+    let output: TransformedSource | string = ''
```

Comment:
not sure what to set here as default return for `process`. Is it possible that `descriptor.script` or `descriptor.scriptSetup` cannot be null/undefined ?

Reply:
Yep, both can be `undefined`.

You can define a vue file like this:
```
<template>
  <div />
</template
```

Without script.